docs/proposals/dedicated-shipwright-operator.md

```diff
@@ -0,0 +1,212 @@
+
+### Risks and Mitigations
+
+**Risk**: The operator provides one of potentially multiple avenues admins can install Shipwright.
+
+*Mitigation*: Documentation will need to provide instructions on supported installation methods.
+
+**Risk**: Manifests uses to deploy Shipwright Builds in the `shipwright-io/operator` are not synchronized with content in `shipwright-io/build`.
+
+*Mitigation*: The operator's CI should include a test suite that runs the e2e tests in `shipwright-io/build`.
+Project maintainers should also ensure that deployment changes to `shipwright-io/build` carry accross to the operator.
+
+## Design Details
+
+### Test Plan
+
+Test suites will need to ensure the following:
+
+1. When a `ShipwrightBuild` object is created, the corresponding build controllers are deployed and the Build APIs are added as custom resource definitions.
+2. Changes to the `ShipwrightBuild` spec are correctly reflected in the subsequent deployment.
+3. CI for the operator should include the e2e suite for `shipwright-io/build`, run against the controllers deployed by the operator.
+
+### Graduation Criteria
+
+##### Dev Preview -> Tech Preview
+
+- Configuration API for `ShipwrightBuild` reaches v1beta1 stability.
+- Installation instructions and configuration options are well documented.
+- Support for basic installation
+
+##### Tech Preview -> GA
+
+- Configuration API for `ShipwrightBuild` reaches v1 stability.
+- Support for over the air upgrades of the operator _and_ the build controllers.
+- [optional] allow version skews between the operator and deployed version of Shipwright Build
+
+### Upgrade / Downgrade Strategy
+
+The operator should use leader election to ensure that when a new version of the operator is installed, it does not conflict with the existing installation.
+This is only required for Tech Preview - Dev Preview releases can assume that the operator is uninstalled before the new version is installed.
+
+## Implementation History
+
+2020-02-03: Proposal
+
+## Drawbacks
+
+A separate operator adds overhead to the project, particularly with respect to synchronizing deployment manifests.
+This can also add confusion if `shipwright-io/build` is made available via a Helm chart or "all in one" Kubernetes manifest.
+
+To avoid confusion, we will also need to rename components in `shipwright-io/build` to remove references to "operator."
+For instance, what we call the `build-operator` today should be renamed the `build-controller-manager`.
+
+## Alternatives
+
+There are other ways to simply install a project like Shipwright:
+
+1. An install script (current approach)
+2. An "all in one" Kubernetes YAML manifest
+3. A Helm chart
+
+However, these mechanisms are one-way installations.
+Even Helm does not include mechanisms to ensure the applied chart is healty and functioning as expected.
+Helm also will not install/upgrade CRDs if they are already present on the cluster.
```

Comment:
👍 Well noted.

To achieve the same admin experience, one would need two helm charts to get around the limitations helm
Has around management of CRDs. - definitely not ideal.